docs/data/tree-view/simple-tree-view/customization/CustomContentTreeView.js

```diff
@@ -1,84 +1,70 @@
 import * as React from 'react';
-import clsx from 'clsx';
 import { styled } from '@mui/material/styles';
-import Typography from '@mui/material/Typography';
 import Box from '@mui/material/Box';
 import Avatar from '@mui/material/Avatar';
 import { SimpleTreeView } from '@mui/x-tree-view/SimpleTreeView';
-import { TreeItem, useTreeItemState } from '@mui/x-tree-view/TreeItem';
+import { useTreeItem } from '@mui/x-tree-view/internals/useTreeItem';
+import {
+  TreeItemNextContent,
+  TreeItemNextIconContainer,
+  TreeItemNextTransitionGroup,
+  TreeItemNextLabel,
+  TreeItemNextRoot,
+} from '@mui/x-tree-view/internals/TreeItemNext';
+import { TreeItemIcon } from '@mui/x-tree-view/internals/TreeItemIcon';
+import { TreeItemProvider } from '@mui/x-tree-view/internals/TreeItemProvider';
 
-const CustomContentRoot = styled('div')(({ theme }) => ({
-  '&': { padding: theme.spacing(0.5, 1) },
+const CustomTreeItemContent = styled(TreeItemNextContent)(({ theme }) => ({
+  padding: theme.spacing(0.5, 1),
 }));
 
-const CustomContent = React.forwardRef(function CustomContent(props, ref) {
+const CustomTreeItem = React.forwardRef(function CustomTreeItem(props, ref) {
   const {
-    className,
-    classes,
-    label,
+    id,
     nodeId,
-    icon: iconProp,
-    expansionIcon,
-    displayIcon,
+    label,
+    children,
+    slots = {},
+    slotProps = {},
+    ...other
   } = props;
 
   const {
-    disabled,
-    expanded,
-    selected,
-    focused,
-    handleExpansion,
-    handleSelection,
-    preventSelection,
-  } = useTreeItemState(nodeId);
-
-  const icon = iconProp || expansionIcon || displayIcon;
-
-  const handleMouseDown = (event) => {
-    preventSelection(event);
-  };
-
-  const handleClick = (event) => {
-    handleExpansion(event);
-    handleSelection(event);
-  };
+    getRootProps,
+    getContentProps,
+    getIconContainerProps,
+    getLabelProps,
+    getGroupTransitionProps,
+    status,
+  } = useTreeItem({ id, nodeId, children, label, rootRef: ref });
 
   return (
-    <CustomContentRoot
-      className={clsx(className, classes.root, {
-        'Mui-expanded': expanded,
-        'Mui-selected': selected,
-        'Mui-focused': focused,
-        'Mui-disabled': disabled,
-      })}
-      onClick={handleClick}
-      onMouseDown={handleMouseDown}
-      ref={ref}
-    >
-      <div className={classes.iconContainer}>{icon}</div>
-      <Box sx={{ flexGrow: 1, display: 'flex', gap: 1 }}>
-        <Avatar
-          sx={(theme) => ({
-            background: theme.palette.primary.main,
-            width: 24,
-            height: 24,
-            fontSize: '0.8rem',
-          })}
-        >
-          {label[0]}
-        </Avatar>
-        <Typography component="div" className={classes.label}>
-          {label}
-        </Typography>
-      </Box>
-    </CustomContentRoot>
+    <TreeItemProvider nodeId={nodeId}>
```

Comment:
This new provider is useful on `SimpleTreeView` to register the JSX items at the tree view level.
And in the future it will also contain potential drag&drop context if needed of stuff like that.